src/lib/support/Pool.h

```diff
@@ -160,4 +167,107 @@ class BitMapObjectPool : public StaticAllocatorBitmap
     } mData;
 };
 
+#if CHIP_SYSTEM_CONFIG_POOL_USE_HEAP
+
+/**
+ *  @brief
+ *      A class template used for allocating Object subclass objects from an ObjectArena<> template union.
+ *
+ *  @tparam     T   a class to be allocated from the arena.
+ */
+template <class T>
+class HeapObjectPool
+{
+public:
+    template <typename... Args>
+    T * CreateObject(Args &&... args)
+    {
+        std::lock_guard<std::mutex> lock(mutex);
```

Comment:
Do we need to lock around the object creation?  Or just around the mObjects access?

Reply:
Mutex removed since the SDK is internally single-thread.